lib/modifiers/batch.js

```diff
@@ -0,0 +1,171 @@
+var through = require('through2');
+var _ = require('lodash');
+
+/**
+ * Returns a new stream that batches writes based on a specific interval, count of items, or both.
+ * All items read on the stream and will be re-emitted as part of an array of one or more items
+ * based on the criteria defined in the options.
+ *
+ * @method
+ * @static
+ * @since    1.5.0
+ * @category Util
+ *
+ * @param   {Object} [options]       - Define how items will be batched and written to the
+ *                                     the stream. Note that when both `time` and `count` are
+ *                                     defined, items will be written to the stream whenever
+ *                                     either one of the conditions is met.
+ * @param   {Number} [options.time]  - The interval, in milliseconds, to limit writes to.
+ *                                     During that time, all items read on the stream will
+ *                                     be collected and written as an array at the set
+ *                                     interval.
+ * @param   {Number} [options.count] - The number of items to buffer before writing all
+ *                                     of them in an array.
+ * @returns {Stream}                 - Transform stream.
+ *
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.batch({ time: 100 }));
+ *
+ * input.write(1);
+ *
+ * setTimeout(function() {
+ *   input.write(2);
+ *   input.write(3);
+ * }, 100);
+ *
+ * setTimeout(function() {
+ *   input.write(4);
+ *   input.write(5);
+ * }, 200);
+ *
+ * // => [[1], [2, 3], [4, 5]]
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.batch({ count: 2 }));
+ *
+ * input.write(1);
+ * input.write(2);
+ * input.write(3);
+ * input.write(4);
+ * input.write(5);
+ *
+ * // => [[1, 2], [3, 4], [5]]
+ */
+function batch(options) {
+  var settings = _.extend({
+    time: 0,
+    count: 0
+  }, options);
+
+  var cache = [];
+  var lastWrite = 0;
+  var timeout;
+
+  /**
+   * @private
+   * @description
+   * Writes all the collected data to the stream.
+   *
+   * @param {Stream} stream - The stream to write to.
+   * @returns {undefined}
+   */
+  function flushCache(stream) {
+    lastWrite = Date.now();
+
+    if (timeout) {
+      clearTimeout(timeout);
+      timeout = null;
+    }
+
+    stream.push(cache.splice(0, settings.count || cache.length));
+  }
+
+  /**
+   * @private
+   * @description
+   * Flushes the cache using the defined timeout.
+   *
+   * @param {Stream} stream    - The stream to write to.
+   * @param {Function} onFlush - The function to call when
+   *                             the cache is flushed.
+   * @returns {undefined}
+   */
+  function flushCacheTimed(stream, onFlush) {
+    var nextWrite = lastWrite + settings.time;
+    var now = Date.now();
+
+    // we are overdue for a write, so write now
+    if (now > nextWrite) {
+      flushCache(stream);
+      onFlush();
+
+      return;
```

Comment:
According to the last travis build, these lines aren't covered. Why is that? https://travis-ci.org/stezu/node-stream/jobs/192553238#L938